modules/drivers/mongo/src/metabase/driver/mongo/query_processor.clj

```diff
@@ -801,6 +819,65 @@
   (compile-cond (negate subclause)))
 
 
+;;; ----------------------------------------------------- joins ------------------------------------------------------
+
+(defn- find-source-table-id [join-or-query]
+  (or (some-> join-or-query :source-table)
+      (some-> join-or-query :source-query recur)))
+
+(defn- find-source-collection [join]
+  (or (-> join :source-query :collection)
+      (-> join find-source-table-id qp.store/table :name)))
+
+(defn- localize-join-alias [expr alias]
+  (mbql.u/replace expr
+    [:field id-or-name (m :guard (every-pred map? #(= (:join-alias %) alias)))]
```

Comment:
I always forget that you can do this, but you can actually match a map like this

```clj
(defn x [e]
  (let [field-id   1
        join-alias "x"]
    (mbql.u/match-one e
      [:field field-id {:join-alias join-alias}]
      :match)))

(x [:field 1 {:join-alias "x"}])
;; => :match

(x [:field 1 {:join-alias "x", :a 1}])
;; => :match

(x [:field 1 {:join-alias "y", :a 1}])
;; => nil
```